src/serialization/base64.cpp

```diff
@@ -0,0 +1,90 @@
+#include "serialization/base64.hpp"
+
+#include <vector>
+
+namespace {
+const std::vector<int>& decode_map(){
+	static std::vector<int> T(256, -1);
+	if(T['A'] == -1){
+		for(int i=0; i<64; i++) {
+			T["ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"[i]] = i;
+		}
+	}
+	return T;
+}
+const std::string itoa64_map = "./0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
```

Comment:
Like I said in another code review, this is logically an array of characters and not a string. Therefore its type should be `const char[]`.

Reply:
The two original places for this string also had it as `std::string`. While I did change the encode function some, I've mostly just moved code to a different place.